gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/documentation/PageQueryServiceImpl.java

```diff
@@ -110,6 +110,23 @@ public long countByParentIdAndIsPublished(String parentId) {
         }
     }
 
+    @Override
+    public Optional<Page> findByNameAndReferenceId(String name, String referenceId) {
+        try {
+            var result = pageRepository.search(new PageCriteria.Builder().name(name).referenceId(referenceId).build());
+            if (!result.isEmpty()) {
+                if (result.size() == 1) {
+                    return Optional.of(PageAdapter.INSTANCE.toEntity(result.get(0)));
+                }
+                throw new IllegalStateException("Found more than one page with name " + name);
+            }
```

Comment:
```suggestion
            var result = pageRepository.search(new PageCriteria.Builder().name(name).build());
            retrun switch (result.size()) {
                case 0 -> Optional.empty();
                case 1 -> Optional.of(PageAdapter.INSTANCE.toEntity(result.get(0)));
                default -> throw new IllegalStateException("Found more than one page with name " + name);
            };
```

For better readability

Reply:
I like that 👍 Good idea, fixed